lib/auth-middleware/auth_middleware/__init__.py

```diff
@@ -220,7 +223,8 @@ def verify_userauth(id_token, group_name=None, internal_token_only=False):
                 logger.warning("bad status getting keyset. status code = %s" % keyset_resp.status_code)
                 raise OAuthProblem('Invalid token')
             keyset = keyset_resp.json()
-            target_client_id = os.getenv('SHIBBOLETH_CLIENT_ID')
+            target_client_id_string = os.getenv('SHIBBOLETH_CLIENT_ID')
+            target_client_id_list = re.split(',+', target_client_id_string)
```

Comment:
Could this be simplified by using only one variable instead of the separate string and list?
```suggestion
            target_client_ids = re.split(',+', target_client_id_string)
```

In line numbers, 244-257, then a loop can be used to remove all blank spaces for each item in `target_client_ids` and then the `target_client_ids` can be used in the jwt.decode method directly.

What do you think?

Reply:
I totally forgot about that jwt can do multiple client. I updated the code